mkt/webapps/tasks.py

```diff
@@ -323,13 +323,23 @@ def update_supported_locales(ids, **kw):
 @use_master
 def index_webapps(ids, **kw):
     # DEPRECATED: call WebappIndexer.index_ids directly.
-    WebappIndexer.index_ids(ids, no_delay=True)
+    homescreens = set(
+        Webapp.tags.through.objects.filter(
+            webapp_id__in=ids,
+            tag__tag_text='homescreen')
+        .values_list('webapp_id', flat=True))
+    webapps = set(ids) - homescreens
+    if homescreens:
+        HomescreenIndexer.index_ids(list(homescreens), no_delay=True)
+    if webapps:
+        WebappIndexer.index_ids(list(webapps), no_delay=True)
```

Comment:
Urgh.

There is got to be a better way. This task is supposed to be deprecated as `obj.get_indexer().index_ids([obj.pk])` does the same job for single objects, and we almost never massively index multiple webapps. Maybe it's a good time to remove it entirely ? If you are dealing with single objects it becomes much easier.